src/averageAges.js

```diff
@@ -1,61 +1,51 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const isMan = (person) => person.sex === 'm';
+const calculateAge = (person) => person.died - person.born;
+const calculateAvgAge = (dates) => dates.reduce((date1, date2) =>
+  date1 + date2) / dates.length;
+const isMother = (child, mother) => child.mother === mother.name;
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => isMan(person));
+  const menAges = century
+    ? men.filter(person => Math.ceil(person.died / 100) === century)
+      .map(person => calculateAge(person)) : men.map(person => (
+      calculateAge(person)));
+
+  const averageAges = calculateAvgAge(menAges);
+
+  return averageAges;
```

Comment:
you can skip variable declaration when you're returning it next-line
```suggestion
  return calculateAvgAge(menAges);
```